setup.py

```diff
@@ -1,16 +1,48 @@
 import os
 from setuptools import setup, Extension
-from Cython.Build import cythonize
-import numpy
+from distutils.command.sdist import sdist as _sdist
+
+try:
+    import numpy
+except ImportError:
+    use_numpy = False
+else:
+    use_numpy = True
+
+try:
+    from Cython.Build import cythonize
+except ImportError:
+    use_cython = False
+else:
+    use_cython = True
 
 name = "ndsplines"
+extension = '_bspl'
+
+cmdclass = {}
+
+if use_cython and use_numpy:
+    extensions = cythonize([
+        Extension("{}.{}".format(name, extension),
+                  [os.path.join(name, "{}.pyx".format(extension))],
+                  include_dirs=[numpy.get_include()],
+                  depends=[os.path.join(name, "{}.h".format(extension))]),
+    ])
 
-extensions = [
-    Extension(f"{name}._bspl",
-              [os.path.join(name, "_bspl.pyx")],
-              include_dirs=[numpy.get_include()],
-              depends=[os.path.join(name, "_bspl.h")]),
-]
+    class sdist(_sdist):
+        def run(self):
+            cythonize([os.path.join(name, "{}.pyx".extension())])
```

Comment:
Ok I just used format strings. I also changed this (used to be `cythonize(['cython/mycythonmodule.pyx'])`) which I think just got left over from copying the template.

Reply:
Also just realized I don't like using the name `extension` and I screwed this one up. Should be fixed now